### DISCUSSION TITLE
In librdkafka how do I tell if a kafka cluster is currently available?

### DISCUSSION THREAD

**Discussion by marlowa** [edenhill/librdkafka#4393] [Q&A]
I have a Kafka producer which receives delivery reports (dr) via the librdkafka callback mechanism. However, I've noticed that when the Kafka cluster goes away my application blats messages to the console (via librdkafka) which my app does not pick up and the dr callback does not get invoked. The calls to produce also apparently succeed so the application doesn't get to find out that kafka is down. Now I know that the produce is mainly enqueueing the message to be picked up by the kafka polling thread and I am guessing that the kafka polling thread considers a loss of connection to be temporary, e.g when a node in the cluster goes down, forcing an election, but what about when the whole cluster is down? I don't seem to get any notify about that. I want to see an API call that allows me to check on the status of a kafka cluster, but I don't see one. Is there one? How do I detect this problem in my code please?

**Comment by rolandyoung** [comment]
Did you see this FAQ article: https://github.com/confluentinc/librdkafka/wiki/FAQ#i-want-to-get-an-event-when-brokers-come-up-and-down ?

The app I maintain monitors the size of the output queue (`rd_kafka_outq_len()`) - if it hits a configurable limit, that is treated as indicating a problem, if it starts to go down again, the problem is considered to have cleared.
